pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/ValidationManager.java

```diff
@@ -122,71 +109,80 @@ public void runValidation() {
     boolean countHLCSegments = true;  // false if this table has ONLY LLC segments (i.e. fully migrated)
     StreamConfig streamConfig = new StreamConfig(tableConfig.getIndexingConfig().getStreamConfigs());
     if (streamConfig.hasSimpleKafkaConsumerType() && !streamConfig.hasHighLevelKafkaConsumerType()) {
       countHLCSegments = false;
     }
     // Update the gauge to contain the total document count in the segments
     _validationMetrics.updateTotalDocumentCountGauge(tableConfig.getTableName(),
-          computeRealtimeTotalDocumentInSegments(metadataList, countHLCSegments));
+        computeRealtimeTotalDocumentInSegments(metadataList, countHLCSegments));
   }
 
   // For offline segment pushes, validate that there are no missing segments, and update metrics
-  private void validateOfflineSegmentPush(ZkHelixPropertyStore<ZNRecord> propertyStore, String offlineTableName) {
+  private void validateOfflineSegmentPush(TableConfig tableConfig) {
+    String offlineTableName = tableConfig.getTableName();
     List<OfflineSegmentZKMetadata> offlineSegmentZKMetadataList =
-        ZKMetadataProvider.getOfflineSegmentZKMetadataListForTable(propertyStore, offlineTableName);
+        _pinotHelixResourceManager.getOfflineSegmentMetadata(offlineTableName);
 
-    // Compute the missing segments if there are at least two
+    // Compute the missing segments if there are at least two segments
     int numMissingSegments = 0;
     int numSegments = offlineSegmentZKMetadataList.size();
-    if (numSegments >= 2) {
+    SegmentsValidationAndRetentionConfig validationConfig = tableConfig.getValidationConfig();
+    if (numSegments >= 2 && StringUtils.isNotEmpty(validationConfig.getTimeColumnName())) {
       List<Interval> segmentIntervals = new ArrayList<>(numSegments);
-      Duration timeGranularity = null;
+      List<String> segmentsWithInvalidInterval = new ArrayList<>();
       for (OfflineSegmentZKMetadata offlineSegmentZKMetadata : offlineSegmentZKMetadataList) {
         Interval timeInterval = offlineSegmentZKMetadata.getTimeInterval();
         if (timeInterval != null && TimeUtils.timeValueInValidRange(timeInterval.getStartMillis())
             && TimeUtils.timeValueInValidRange(timeInterval.getEndMillis())) {
           segmentIntervals.add(timeInterval);
-          timeGranularity = offlineSegmentZKMetadata.getTimeGranularity();
+        } else {
+          segmentsWithInvalidInterval.add(offlineSegmentZKMetadata.getSegmentName());
         }
       }
-      List<Interval> missingIntervals = computeMissingIntervals(segmentIntervals, timeGranularity);
-      for (Interval missingInterval : missingIntervals) {
-        LOGGER.warn("Missing data in table {} for time interval {}", offlineTableName, missingInterval);
+      if (!segmentsWithInvalidInterval.isEmpty()) {
+        LOGGER.warn("Table: {} has segments with invalid interval: {}", offlineTableName, segmentsWithInvalidInterval);
```

Comment:
I am wondering what the utlity of this log message is, given that we already set a metric with the number of missing intervals. it could also end up being a very large log message.

If the metric is non-zero, we can list the segments of the table and detect the missing segments (if the owner of the table so desires).

In any case, I am not sure this is a WARN. It should be INFO? (we will log WARN and ERROR messages if a segment push failed on the pinot controller, I suppose).

Reply:
I would prefer WARN because this message denotes some abnormal for the table.
If we don't need the interval for missing segments, maybe we can just log the number of missing segments. What's your opinion on this?

Reply:
A WARN indicates something is wrong with pinot (controller). That is not necessarily the case here (I know that it _can_ be the case and hence the question mark on should it be an info). This can be due to a failure while pushing segment on the offline side (e.g. hadoop) and pinot never got the segment. It seems to me that info is enough. 

Also, we can probably remove the log altogether, since we already have a metric and can monitor it. If we list the segments we can find the missing intervals.

Reply:
IMO, INFO is purely for debugging purpose(we don't enable DEBUG usually), WARN is something we should keep an eye on (not necessary something goes wrong), ERROR means something unexpected happens, which probably needs bug fix.
For the log, if we got too many tables, the table name might be replaced with "allTable" to reduce the number of metrics (not happening right now for ValidationMetrics). I prefer at least keep a one line log for number of missing segments.